src/Sql/Sql/ManagedDatabase/Cmdlet/MoveCopy/GetMoveCopyAzureSqlManagedDatabaseOperationsBase.cs

```diff
@@ -0,0 +1,105 @@
+        [Parameter(Mandatory = false, HelpMessage = "The name of the target resource group.", ParameterSetName = GetMoveCopyManagedDatabaseOperationsByResourceIdParameterSet)]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, ParameterSetName = GetMoveCopyManagedDatabaseOperationsByMoveCopyObjectParameterSet, HelpMessage = "The name of the target resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string TargetResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The name of the target managed instance.", ParameterSetName = GetMoveCopyManagedDatabaseOperationsByNameParameterSet)]
+        [Parameter(Mandatory = false, HelpMessage = "The name of the target managed instance.", ParameterSetName = GetMoveCopyManagedDatabaseOperationsByInputObjectParameterSet)]
+        [Parameter(Mandatory = false, HelpMessage = "The name of the target managed instance.", ParameterSetName = GetMoveCopyManagedDatabaseOperationsByResourceIdParameterSet)]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, ParameterSetName = GetMoveCopyManagedDatabaseOperationsByMoveCopyObjectParameterSet, HelpMessage = "The name of the target managed instance.")]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter("Microsoft.Sql/managedInstances", nameof(TargetResourceGroupName))]
+        public string TargetInstanceName { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = GetMoveCopyManagedDatabaseOperationsByInputObjectParameterSet)]
+        [ValidateNotNull]
+        public AzureSqlManagedDatabaseModel DatabaseObject { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = GetMoveCopyManagedDatabaseOperationsByResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = GetMoveCopyManagedDatabaseOperationsByMoveCopyObjectParameterSet, HelpMessage = "Object that is returned from start copy operation.")]
+        [ValidateNotNullOrEmpty]
+        public MoveCopyManagedDatabaseModel ModelObject { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "It will return only latest opereation per database")]
+        public SwitchParameter OnlyLatestPerDatabase { get; set; }
+
+        /// <summary>
+        /// Get the operation mode of the command. COPY or MOVE
+        /// </summary>
+        /// <returns>Operation mode</returns>
+        protected abstract OperationMode GetOperationMode();
+
+
+        protected override IList<ManagedDatabaseMoveCopyOperation> GetEntity()
+        {
+            if (ParameterSetName.Equals(GetMoveCopyManagedDatabaseOperationsByResourceIdParameterSet))
```

Comment:
do we also need some arg transformation when using `byObject` parameter sets?

Reply:
what if we call it without piping, like 
```
$dbObj123 = Get-AzSqlDb...
Get-AzSqlInstanceDatabaseCopy... -DatabaseObject $dbObj123
```

Reply:
You are correct, forgot about this use-case

Reply:
fixed.